src/ledgerhelpers/journal.py

```diff
@@ -120,7 +120,7 @@ class Journal(JournalCommon):
     slave = None
     slave_lock = None
     cache = None
-    internal_parsing_cache = None
+    internal_parsing_cache = []
```

Comment:
This is not a legitimate use of an empty list, because then it becomes a shared list among journals.

You must leave this at None and either override specifically in the constructor, or change all uses to return an empty list by checking that the cache is not None and its first element is not None.

Reply:
AFAICT no uses of this field seems to mutate this list (it is only replaced, never appended to), but probably better safe than sorry.